components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -598,7 +598,9 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     public async deleteAccount(ctx: TraceContext): Promise<void> {
         const user = this.checkAndBlockUser("deleteAccount");
         await this.guardAccess({ kind: "user", subject: user! }, "delete");
-
+        if (user.blocked) {
+            throw new ResponseError(ErrorCodes.PERMISSION_DENIED, `You are blocked and can't delete your account.`);
```

Comment:
@svenefftinge, that was already done in https://github.com/gitpod-io/gitpod/pull/9372.
the `this.checkAndBlockUser("deleteAccount")` contains the errors to be thrown if `user.blocked === true`

Reply:
Awesome! 🎉